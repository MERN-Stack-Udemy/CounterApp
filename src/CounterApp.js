import React, { useState } from 'react'
import PropTypes from 'prop-types';

function CounterApp( {value = 12} ) {
  
  const [ counter, setCounter ] = useState(value);

  const handdleAdd = () => setCounter( (c) => c + 1 );

  const handdleReset = () => setCounter(value);

  const handdleSub = () => setCounter(counter - 1);


  return (
    <div>
      <h1>Counter App</h1>
      <h2>Value : {counter}</h2>
      <button onClick={ () => handdleSub("SUB") } >-1</button>
      <button onClick={ () => handdleReset("RESET") } >RESET</button>
      <button onClick={ () => handdleAdd("ADD") } >+1</button>
    </div>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}

export default CounterApp
