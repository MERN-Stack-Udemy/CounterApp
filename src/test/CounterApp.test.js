import { shallow } from 'enzyme'
import CounterApp from '../CounterApp'

describe('Test sobre <CounterApp />', () => {

  let wrapper = shallow( <CounterApp value = {10} /> );

  beforeEach( () => {
    wrapper = shallow( <CounterApp value = {10} /> );
  });

  test('snapshot de counterApp', () => {
    
    // const wrapper = shallow( <CounterApp value = {value} /> );
    expect(wrapper).toMatchSnapshot();
  })
  
  test('validar el text del elemento renderizado acorde a `value`', () => {
    const value = 100;
    const renderValueParrafo = `Value : ${value}`
    const wrapper = shallow( <CounterApp value = {value} /> );
    const valueParrafo = wrapper.find('h2').text();
    expect( valueParrafo ).toBe( renderValueParrafo );
  })

  test('debe cambiar el valor en -1 al precionar el boton ', () => {
    wrapper.find('button').at(0).simulate('click');
    const counterText = wrapper.find('h2').text()
    // console.log(counterText)
    expect( counterText ).toBe("Value : 9")
  })
  
  test('debe cambiar el valor en +1 al precionar el boton ', () => {
    wrapper.find('button').at(2).simulate('click');
    const counterText = wrapper.find('h2').text()
    // console.log(counterText)
    expect( counterText ).toBe("Value : 11")
  })

  test('Debe volver al valor por inicial al presionar reset', () => {
    const wrapper = shallow( <CounterApp value = {105} /> );
    
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    let counterText = wrapper.find('h2').text()
    // console.log(counterText);
    wrapper.find('button').at(1).simulate('click');
    counterText = wrapper.find('h2').text()
    expect( counterText ).toBe("Value : 105")
  })
  

})
