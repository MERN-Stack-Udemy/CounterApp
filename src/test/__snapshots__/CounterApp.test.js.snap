// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test sobre <CounterApp /> snapshot de counterApp 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    Value : 
    10
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    RESET
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;
